### PR TITLE
Pub/Sub RemoteForwarder

### PR DESCRIPTION
**Pull request by SanjoDeundiak**

### CODE REVIEW

File: implementations/rust/ockam/ockam_transport_tcp/src/router/mod.rs [hunk 1]
```diff
@@ -120,13 +120,6 @@ impl Worker for TcpRouter {
     type Context = Context;
     type Message = RouterMessage;
 
-    async fn initialize(&mut self, ctx: &mut Context) -> Result<()> {
```
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_transport_tcp/src/router/mod.rs:123]: @spacekookie FYI. This was causing a race condition. Are there other places like this?

File: implementations/rust/ockam/ockam_node/src/heartbeat.rs [hunk 2]
```diff
@@ -0,0 +1,271 @@
+use crate::Context;
```
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_node/src/heartbeat.rs:1]: @spacekookie FYI

File: implementations/rust/ockam/ockam_transport_tcp/src/router/mod.rs [hunk 3]
```diff
@@ -168,6 +161,9 @@ impl TcpRouter {
         let handle = router.create_self_handle(ctx).await?;
 
         ctx.start_worker(addr.clone(), router).await?;
+        trace!("Registering TCP router for type = {}", TCP);
+        ctx.register(TCP, addr).await?;
+        ctx.set_cluster(crate::CLUSTER_NAME).await?;
```
[the remark this answers is not in the document]
Comment (SanjoDeundiak) [reply implementations/rust/ockam/ockam_transport_tcp/src/router/mod.rs:166]: I can move set_cluster call back

[the remark this answers is not in the document]
Comment (SanjoDeundiak) [reply implementations/rust/ockam/ockam_transport_tcp/src/router/mod.rs:166]: Basically, the fact that TcpTransport::create call succeeded, doesn't mean you can use Tcp...

**Review by hairyhum — DISMISSED** [review]

**Review by thomcc — APPROVED** [review]
Seems fine to me.

**Pull request merged**